core/runtime/exception_state.rs

```diff
@@ -0,0 +1,127 @@
+use crate::error::exception_to_err_result;
+use anyhow::Error;
+use std::cell::Cell;
+use std::cell::RefCell;
+use std::collections::VecDeque;
+use std::rc::Rc;
+
+#[derive(Default)]
+pub(crate) struct ExceptionState {
+  // TODO(nayeemrmn): This is polled in `exception_to_err_result()` which is
+  // flimsy. Try to poll it similarly to `pending_promise_rejections`.
```

Comment:
Is this comment still valid?

Reply:
Unfortunately yes 😢 

We should be able to make this work like a waker, but that will need a fair bit of work.